cmd/snap-bootstrap/bootstrap/tpm_test.go

```diff
@@ -0,0 +1,176 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package bootstrap_test
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/chrisccoulson/go-tpm2"
+	"github.com/snapcore/secboot"
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/cmd/snap-bootstrap/bootstrap"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type bootstrapTPMSuite struct {
+	testutil.BaseTest
+
+	dir string
+}
+
+var _ = Suite(&bootstrapTPMSuite{})
+
+func (s *bootstrapTPMSuite) SetUpTest(c *C) {
+	s.BaseTest.SetUpTest(c)
+
+	s.dir = c.MkDir()
+	dirs.SetRootDir(s.dir)
+	s.AddCleanup(func() { dirs.SetRootDir("/") })
+}
+
+func (s *bootstrapTPMSuite) TestProvision(c *C) {
+	n := 0
+	restore := bootstrap.MockProvisionTPM(func(tpm *secboot.TPMConnection, mode secboot.ProvisionMode, newLockoutAuth []byte) error {
+		c.Assert(mode, Equals, secboot.ProvisionModeFull)
+		n++
+		return nil
+	})
+	defer restore()
+
+	t := bootstrap.TPMSupport{}
+	err := t.Provision()
+	c.Assert(err, IsNil)
+	c.Assert(n, Equals, 1)
+}
+
+func (s *bootstrapTPMSuite) TestSeal(c *C) {
```

Comment:
All of this tests a single shim/bootloader/kernel - if we support multiple ones in the API we should test multiple ones here too (or we just support a single one for the time being if that is enough and avoid this extra testing).

Reply:
I see your point. Let's stay with the simple version for now and add the multiple file version back in a different PR when it's required.